### PR TITLE
Add https support to readinessProbe

### LINKED ISSUE #16: Kibana readinessProbe fails when server.ssl.enabled=true
The [readinessProbe](https://github.com/elastic/helm-charts/blob/master/kibana/templates/deployment.yaml#L55-L73) only appears to target HTTP, not HTTPS.

### PR DESCRIPTION
**Pull request by jordansissel**
If server.ssl.enabled is set in Kibana (vi SERVER_SSL_ENABLED env var),
then the readinessProbe will use https to test for readiness.

Fixes #16

### CODE REVIEW

**Comment by jordansissel** [comment]
Caveat: This is my first effort to do anything of substance with Helm.

Open questions:
* Is `kibanaSSLHostname` the best value name?
* Does the shell script need more comments?

**Comment by jordansissel** [comment]
Test results:

```
% kubectl get pods -l app=kibana
NAME                            READY   STATUS    RESTARTS   AGE
xyz-kibana-5dfb9fcdb8-pfv6l   1/1     Running   0          15m
xyz-kibana-5dfb9fcdb8-s6zsq   1/1     Running   0          15m
```

(Without this patch, the `READY` state would forever be `0/1` as the check over http would always fail)

File: kibana/values.yaml [hunk 1]
```diff
@@ -53,6 +53,10 @@ service:
   type: ClusterIP
   port: 5601
 
+# Set this if you are setting server.ssl.enabled in Kibana.
+# This value is a hostname accepted by the SSL certificate provided to kibana.
+kibanaSSLHostname: localhost
```
Comment (Crazybus) [inline kibana/values.yaml]: Could you also this to the [configuration](https://github.com/elastic/helm-charts/tree/master/kibana#configuration) section of the readme?

**Comment by Crazybus** [comment]
> Is kibanaSSLHostname the best value name?

Sounds right to me and it is following the helm variable naming conventions. 

> Does the shell script need more comments?

The code is clear and logical to me, I didn't have any trouble understanding it without comments 👍

File: kibana/values.yaml (hunk 1 above)
Comment (jordansissel) [reply kibana/values.yaml]: +1 will do.

**Comment by Crazybus** [comment]
Thanks for adding your config! I'll update the integration test/example to use it.

**Pull request merged**